### DISCUSSION TITLE
How about the relationship of mixcr gene library and IMGT gene library?

### DISCUSSION THREAD

**Discussion by omegahh** [milaboratory/mixcr#1311] [Q&A] [answered]
As far as I know, the gene names in MIXCR library are different with those in IMGT library. Is there any table or result about comparison of these two libraries?

**Comment by mizraelson** [comment, marked as answer]
Hi

All gene names in the MiXCR internal library are consistent with those in IMGT. It's important to note that, due to MiXCR's reliance on a genome-assembly-based library, the default output clonotype tables only showcase one allelic variant for each gene (the *00 allele). In most situations, this allele corresponds to the *01 allele in IMGT. However, if necessary, you can infer allelic variants in the MiXCR output using the [mixcr findAlleles](https://mixcr.com/mixcr/reference/mixcr-findAlleles/) command.

Opting for this method of library construction offers several advantages over IMGT:

1. Genomic Basis: MiXCR constructs its library using genome assemblies from various species, ensuring a representation that's more in line with genuine genetic information than what's found in the IMGT database.
 
2. Elimination of False Positives: Unlike IMGT, the MiXCR library is tailored to reduce the occurrence of false positive duplicates, thereby elevating the accuracy and dependability of the data provided to researchers.

3. Comprehensive V Gene Sequences: The MiXCR library offers full sequences for V genes. This encompasses non-coding regions like the 5’UTR, the introductory exon (L1), and the L2 sequence – a breadth of coverage that the IMGT database doesn't provide.

4. Full Transcript Alignment: By integrating non-coding regions, the first exon of V genes, and L2, MiXCR's library facilitates alignment to the complete VDJ transcript sequence. This results in a more nuanced differentiation of V genes that are closely related.

5. Validation of Allelic Diversity: MiXCR's human reference library comes equipped with an alleles database, sourced from actual repseq data. This ensures the inclusion of only genuine allelic variants, further bolstering data reliability. This feature is applied apon using mentioned above [mixcr findAlleles](https://mixcr.com/mixcr/reference/mixcr-findAlleles/) command.

Sincerely,
Mark

**Comment by omegahh** [comment]
Thank you for answering. It is very useful for me.

**Comment by PoslavskySV** [comment]
let's keep this discussion open as more questions may arise from the community

**Comment by omegahh** [comment]
I have a new question about the difference between IMGT library and Gene library. For species 'Rat', I found that the IMGT only have IG genes and does not have any TR genes, while Gene library shows the opposite result. I don't know why there is this difference.

<img width="1707" alt="截屏2023-09-25 17 50 48" src="https://github.com/milaboratory/mixcr/assets/5026687/67b2f92c-a076-4a6c-83f0-1210540e9673">

<img width="1680" alt="截屏2023-09-25 17 52 56" src="https://github.com/milaboratory/mixcr/assets/5026687/683d8f80-91d9-4390-8508-2def193126e7">

**Comment by mizraelson** [comment]
Indeed, we haven't yet added all the references present in IMGT to our library. When we construct our reference, we do it from scratch using both genomic and transcriptomic data, aligning with IMGT in terms of gene names when they are available. We are continuously updating MiXCR with new reference libraries, and the ones currently missing will be incorporated soon.
